src/DessserQCheck.ml

```diff
@@ -173,61 +173,45 @@ let shrink_mac_type mt =
   loop to_simplest
 
 let rec shrink_value_type =
+  let open Iter in
   let vt_of_mn mn = mn.T.vtyp in
-  let shrink_fields mns make_typ f =
-    Array.iter (fun (_, mn) -> shrink_maybe_nullable mn (f % vt_of_mn)) mns ;
-    let shrink_mns =
-      let shrink (fn, mn) =
-        Iter.map (fun mn -> fn, mn) (shrink_maybe_nullable mn) in
-      Shrink.filter (fun mns -> Array.length mns > 1)
-        (Shrink.array ~shrink) mns |>
-      Iter.map make_typ in
-    shrink_mns f in
+  let shrink_fields f =
+    (Shrink.pair Shrink.nil shrink_maybe_nullable) f in
   function
-  | T.Unknown | T.Ext _ ->
-      Iter.empty
-  | T.Base Unit ->
-      Iter.empty
-  | T.Base mt ->
-      (fun f ->
-        shrink_mac_type mt (fun mt -> f (T.Base mt)))
-  | T.Usr _ ->
-      Iter.empty
+  | T.Unknown | T.Ext _ -> empty
+  | T.Base Unit -> empty
+  | T.Base mt -> shrink_mac_type mt >|= T.base
+  | T.Usr _ -> empty
   | T.Vec (dim, mn) ->
-      (fun f ->
-        shrink_maybe_nullable mn (fun mn ->
-          f mn.vtyp ;
-          f (T.Vec (dim, mn))))
+      let smn = shrink_maybe_nullable mn in
+      (smn >|= T.vec dim) <+> (smn >|= vt_of_mn)
```

Comment:
I do not think order is important here. Shrink are just iterators. To my understanding `a <+> b` should be equal to `b <+> a`

Reply:
We can keep this as a convention though.

Reply:
Though you have this in the [documentation](https://c-cube.github.io/qcheck/0.17/qcheck-core/QCheck/Shrink/index.html):

```
pair a b uses a to shrink the first element of tuples, then tries to shrink the second element using b. It is often better, when generating tuples, to put the "simplest" element first (atomic type rather than list, etc.) because it will be shrunk earlier. In particular, putting functions last might help.
```

Reply:
Have a look at the code or build a test that shows once and for all if/how is order relevant.